src/Microsoft.ML.StandardLearners/Standard/ModelStatistics.cs

```diff
@@ -222,10 +231,97 @@ public static bool TryGetBiasStatistics(LinearModelStatistics stats, Single bias
             stdError = stats._coeffStdError.Value.Values[0];
             Contracts.Assert(stdError == stats._coeffStdError.Value.GetItemOrDefault(0));
             zScore = bias / stdError;
-            pValue = 1 - (Single)ProbabilityFunctions.Erf(Math.Abs(zScore / sqrt2));
+            pValue = 1.0f - (Single)ProbabilityFunctions.Erf(Math.Abs(zScore / sqrt2));
             return true;
         }
 
+        /// <summary>
+        /// Computes the standart deviation matrix of each of the non-zero training weights, needed to calculate further the standart deviation,
+        /// p-value and z-Score.
+        /// If you need faster calculations, use the ComputeStd method from the Microsoft.ML.HALLearners package, which makes use of hardware acceleration.
+        /// </summary>
+        /// <param name="model">A <see cref="LinearBinaryPredictor"/> obtained as a result of training with <see cref="LogisticRegression"/>.</param>
+        /// <param name="ch">The <see cref="IChannel"/> used for messaging.</param>
+        /// <param name="l2Weight">The L2Weight used for training. (Supply the same one that got used during training.)</param>
+        public static void ComputeStd(LinearBinaryPredictor model, IChannel ch, float l2Weight = LogisticRegression.Arguments.Defaults.L2Weight)
```

Comment:
>ComputeStd [](start = 27, length = 10)

I do dislike it being here, but also putting them in the LinearBinaryPredictor doesn't make sense, since the computations are really only if the model gets generated through LR training (but the LinearBinaryPredictor is the predictor for most of our binary classifiers). @tfinley@gmail.com I am starting to think that the best place for it might indeed be the  LR trainer. Maybe a static method in the LR trainer with the same signature as here.

Reply:
Actually I'll refactor another class out of the LinearBinaryPredictor that will have the Stats object, and use that as the model produced the Lbfgs trainers. 

---
In reply to: [231417895](https://github.com/dotnet/machinelearning/pull/1392#discussion_r231417895) [](ancestors = 231417895)